lib/widgets/button.dart

```diff
@@ -0,0 +1,169 @@
+import 'package:flutter/material.dart';
+
+import 'color.dart';
+import 'text.dart';
+import 'theme.dart';
+
+/// The "Button" component from Zulip Web UI kit.
+///
+/// The Figma uses this for the "Cancel" and "Save" buttons in the compose box
+/// for editing an already-sent message.
+///
+/// See Figma:
+///   * Component: https://www.figma.com/design/msWyAJ8cnMHgOMPxi7BUvA/Zulip-Web-UI-kit?node-id=1-2780&t=Wia0D0i1I0GXdD9z-0
+///   * Edit-message compose box: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3988-38201&m=dev
+class ZulipWebUiKitButton extends StatelessWidget {
+  const ZulipWebUiKitButton({
+    super.key,
+    this.attention = ZulipWebUiKitButtonAttention.medium,
+    this.intent = ZulipWebUiKitButtonIntent.info,
+    required this.label,
+    required this.onPressed,
+  });
+
+  final ZulipWebUiKitButtonAttention attention;
+  final ZulipWebUiKitButtonIntent intent;
+  final String label;
+  final VoidCallback? onPressed;
```

Comment:
Looks like the Figma design does not offer style for disabled buttons. Widgets like `TextButton` are disabled when `onPressed` is `null`, but we don't have special styling to indicate that (and `AnimatedScaleOnTap` still works).